lib/ecl/ecl_sum.c

```diff
@@ -690,6 +702,56 @@ const char * ecl_sum_get_general_var_unit( const ecl_sum_type * ecl_sum , const
 
 /*****************************************************************/
 
+
+ecl_sum_type * ecl_sum_alloc_resample(const char * ecl_case, const ecl_sum_type * ecl_sum, const time_t_vector_type * times) {
+
+  time_t start_time = ecl_sum_data_get_data_start(ecl_sum->data);
+
+  if ( time_t_vector_get_first(times) < start_time )
+    return NULL;
+  if ( time_t_vector_get_last(times) > ecl_sum_data_get_sim_end(ecl_sum->data) )
+    return NULL;
+  if ( !time_t_vector_is_sorted(times, false) )
+    return NULL;
+  
+  bool   fmt_case        = ecl_sum->fmt_case;
```

Comment:
Not necessary to create local variables for these - just use `ecl_sum->fmt_case`, `ecl_sum->unified` and so on in the `ecl_sum_alloc_writer( )` call.